doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,281 @@
+### 2.2. Monitoring and Thermal Management
+* Dpu State
+    * The DPUs should provide their state to the host as the boot progression happens, through the host hardware (see the DPU state table for details)
+    * SmartSwitch should store the dpu state data in the DPU_STATE table in the host ChassisStateDB (explained in DB schema)
+    * DPUs should be able to store the data using a redis call
+    * The DPU must provide additional information on the state once it boots its OS to DPU_STATE table.
+    * The SmartSwitch host PMON should be able to monitor the liveliness of the DPUs and when they go down should be able to take appropriate actions such as updating the state of the DPU in the DB and should try to gracefully recover the DPU when requested by the PMON
+
+* Thermal management
+    * Sensor values and fan speeds, status should be read periodically and stored in SmartSwitch StateDB
+    * Platform modules should use this thermal sensor values against the thresholds in the thermal policy and adjust fan speeds depending on the temperature
+    * Trigger thermal shut down on critical policy violation
+
+* Show CLIs
+    * Extend existing CLIs such as show platform fan/temperature to support the new HW
+    * Add new CLIs 
+        ```
+        show dpu <id> state
+        ```
+### 2.3. Detect and Debug
+* Health
+    * SmartSwitch DPUs should store their health data in their local StateDB 
+    * DPUs should support a CLI to display the health data “show dpu health”
+    * The host should be able to access this data using a redis call or an api
+* Alarm and Syslog
+    * Raise alarms when the temperature thresholds exceed, fans run slow or not present or faulty
+    * Drive LEDs accordingly
+    * Provide LED status indicators for DPU boards
+    * Trigger syslog
+* Console
+    * Provide console access to the DPUs through the Host CPU from the front panel management port
+### 2.3. RMA
+* The dpu-cards should be displayed as part of inventory
+* Extend the CLI “show platform inventory” to display the dpu-cards and their state
+* The system should be powered down for replacement of dpu-card (SLED)
+
+## 3.	SmartSwitch PMON Design
+SmartSwitch PMON block diagram
+<p align="center"><img src="./images/pmon-blk-dgm.svg"></p>
+
+### 3.1. Platform monitoring and management
+* SmartSwitch design Extends the existing chassid
+* Extend chassisd for dpu and add dpu-crad as a module besides the existing rp, lc, fc
+* Abstract dpu-card (SLED) for higher level constructs as needed (Ex: Provide APIs at DPU level shown below)
+* changes to platform plugin
+    * extend module.py to support dpu module
+        * class DpuCardModule(Module):
+    * Update the DPU_STATE table information read from platform (refer DB schema section)
+    * Once DPU boots it will update the remaining DPU_STATE information in the SmartSwitch ChassisStateDB
+* APIs
+    * Existing APIs: get_name, get_type, get_my_slot_idx, etc
+    * New APIs
```

Comment:
use these existing APIs
- get_oper_status(self)
- get_module_index(self, module_name)
- get_midplane_ip(self):
- get_base_mac(self)

Note: there is no notion of slot for DPU, rather use port number

Reply:
Outdated